rclcpp/include/rclcpp/node.hpp

```diff
@@ -481,6 +489,81 @@ class Node : public std::enable_shared_from_this<Node>
   rclcpp::node_interfaces::NodeTimeSourceInterface::SharedPtr
   get_node_time_source_interface();
 
+  /// Return the sub-namespace, if this is a sub-node, otherwise an empty string.
+  /**
+   * The returned sub-namespace is what was given to create_sub_node() or an
+   * empty string if this is an original node instance, i.e. not a sub-node.
+   *
+   * get_namespace() will return the original node namespace, and will not
+   * include the sub-namespace if one exists.
+   * To get that you need to call the get_effective_namespace() method.
+   *
+   * \sa get_namespace()
+   * \sa get_effective_namespace()
+   * \return the sub-namespace string, not including the node's original namespace
+   */
+  RCLCPP_PUBLIC
+  const std::string &
+  get_sub_namespace() const;
+
+  /// Return the effective namespace that is used when creating entities.
+  /**
+   * The returned namespace is a combination of the node namespace and the
+   * accumulated sub-namespaces, which is used as the namespace when creating
+   * entities which have relative names.
+   *
+   * \sa get_namespace()
+   * \sa get_sub_namespace()
+   * \return the sub-namespace string, not including the node's original namespace
+   */
+  RCLCPP_PUBLIC
+  const std::string &
+  get_effective_namespace() const;
+
+  /// Create a sub-node, which will extend the namespace of all entities created with it.
+  /**
+   * A normal node will not have a sub-namespace (short for subordinate
+   * namespace) and therefore entities created will be unaffected.
+   * However, if a sub-node (short for subordinate node) is created, then
+   * its sub-namespace will be extended with the sub-namespace given here, and
+   * any entities created with the sub-node object instead of a normal node
+   * object will be affected by the sub-namespace.
```

Comment:
I'm finding the above paragraph a little difficult to read, partially because it is described in the "negative" sense, partially because it seems to be more like high-level documentation rather than documentation for this particular API, and partially because the terms aren't all defined.  I might suggest something more like the following:

```
Any entities (topics, services, etc.) created with this Node pointer will be extended with the sub-namespace (short for subordinate namespace) given here.  Note that entities created using a "normal" Node pointer will not be affected by this sub-namespace.
```

Reply:
I can work on the wording, but in your suggestion it's not necessarily a pointer, I think instance is a more appropriate term, and I don't see the reason to avoid the `sub-node` term, as it is described (defined?) in the `create_sub_node()` method.

Reply:
I rewrote this section in the doc block, please take another look.

I have so much in this particular doc block because I don't know of a better place to describe these concepts right now.